### PR TITLE
Gio receivepage

### PR DESCRIPTION
**Pull request by Sirmorrison**
### Resolves

Issue #7

### What's new

Added the receive page ui with dummy data

### Relevant screenshots or logs

landing ui
image

image

image

image

### Additional context

NA

### CODE REVIEW

File: ui/recieve_page.go [hunk 1]
```diff
@@ -1 +1,317 @@
...
+	addrs       string
+
+	isInfoBtnModal = false
+	isNewAddrModal = false
+
+	selectedWallet  *wallet.InfoShort
+	selectedAccount *wallet.Account
+
+	pageTitle       = "Receiving DCR"
+	ReceivePageInfo = "Each time you request a payment, a \nnew address is created to protect \nyour privacy."
```
Comment (oluwandabira) [inline ui/recieve_page.go]: All this should be part of `Window`

File: ui/recieve_page.go [hunk 2]
```diff
@@ -1 +1,317 @@
...
+
+	var addr string
+	var err error
+
+	// create a new receive address everytime a new account is chosen
+	if generateNew {
+		addr, err = win.wallet.NextAddress(wallet.ID, account.Number)
+		if err != nil {
+			win.outputs.err.Text = err.Error()
+			return
```
Comment (oluwandabira) [inline ui/recieve_page.go]: Why do you need to generate a new address?

File: ui/widgets.go [hunk 3]
```diff
@@ -13,8 +15,8 @@ type inputs struct {
 	createWallet, restoreWallet, deleteWallet                 widget.Button
 	toOverview, toWallets, toTransactions, toSend, toSettings widget.Button
 	toRestoreWallet                                           widget.Button
-	//toReceive                                                 widget.Button
-	sync widget.Button
+	toReceive                                                 widget.Button
+	sync, info, more, dropdown, copy, gotIt, newAddress       widget.Button
```
Comment (oluwandabira) [inline ui/widgets.go]: `gotIt` ? really?

Comment (oluwandabira) [inline ui/widgets.go]: I suggest moving these to a substruct of widgets just for the receive page

File: wallet/responses.go [hunk 4]
```diff
@@ -36,23 +36,34 @@ type MultiWalletInfo struct {
...
-	BlockTimestamp  int64
-	IsWaiting       bool
+	ID               int
+	Name             string
+	Balance          string
+	Accounts         []Account
+	BestBlockHeight  int32
+	BlockTimestamp   int64
+	IsWaiting        bool
+	TotalBalance     int64
```
Comment (oluwandabira) [inline wallet/responses.go:51]: Why is this here when there is `Balance`?

File: wallet/responses.go [hunk 5]
```diff
@@ -36,23 +36,34 @@ type MultiWalletInfo struct {
...
 type Account struct {
-	Number         string
-	Name           string
-	TotalBalance   string
-	CurrentAddress string
+	Number           int32
+	Name             string
+	TotalBalance     string
+	CurrentAddress   string
+	SpendableBalance int64
```
Comment (oluwandabira) [inline wallet/responses.go:75]: Should be a string if it's only going to be displayed

**Review by oluwandabira — CHANGES_REQUESTED** [review]

File: ui/recieve_page.go (hunk 2 above)
Comment (Sirmorrison) [reply ui/recieve_page.go]: you may need to be more explicit please as ion follow.

Comment (oluwandabira) [reply ui/recieve_page.go]: Why do you need to call `NextAddress` every time a new account is chosen? The current one would remain unused then.

Comment (Sirmorrison) [reply ui/recieve_page.go]: so what do you recommend?

File: ui/recieve_page.go [hunk 6]
```diff
@@ -1 +1,301 @@
...
+)
+
+var (
+	listContainer = &layout.List{Axis: layout.Vertical}
+)
+
+func (win *Window) Receive() {
+	if win.walletInfo.LoadedWallets != 0 {
+		win.setDefaultPageValues()
+	}
```
Comment (oluwandabira) [inline ui/recieve_page.go]: Show no wallets loaded like with the wallets page.

File: ui/recieve_page.go [hunk 7]
```diff
@@ -1 +1,301 @@
...
+	})
+	if win.isNewAddrModal {
+		win.drawMoreModal()
+	}
+	if win.isInfoBtnModal {
+		win.drawInfoModal()
+	}
+}
+
+func (win *Window) pageFirstColumn() {
```
Comment (oluwandabira) [inline ui/recieve_page.go]: Name is too generic

File: ui/widgets.go [hunk 8]
```diff
@@ -34,21 +39,28 @@ type combined struct {
...
-	createWallet, restoreDiag, restoreWallet, deleteWallet, deleteDiag decredmaterial.Button
-	sync                                                               decredmaterial.IconButton
+	createWallet, restoreDiag, restoreWallet, deleteWallet, deleteDiag, gotItDiag decredmaterial.Button
+	newAddress                                                                    decredmaterial.Button
+	sync, info, more, copy                                                        decredmaterial.IconButton
 
-	tabs                          []decredmaterial.TabItem
-	notImplemented, noWallet, err decredmaterial.Label
+	tabs                                                      []decredmaterial.TabItem
+	notImplemented, noWallet, receivePageTitle, pageInfo, err decredmaterial.Label
```
Comment (oluwandabira) [inline ui/widgets.go]: `reveivePageTitle` -> `pageTitle`
I'm not even sure we need page titles.

File: ui/window.go [hunk 9]
```diff
@@ -22,10 +22,18 @@ type Window struct {
...
+	selectedAccount *wallet.Account
+
 	current layout.Widget
 	dialog  layout.Widget
 	tabs    *decredmaterial.Tabs
 
+	addrs string
+
+	isInfoBtnModal bool
+	isNewAddrModal bool
```
Comment (oluwandabira) [inline ui/window.go]: Remove. There is already a state for if the window is showing a modal

File: ui/window.go [hunk 10]
```diff
@@ -22,10 +22,18 @@ type Window struct {
...
 	walletInfo *wallet.MultiWalletInfo
 
+	selectedWallet  *wallet.InfoShort
+	selectedAccount *wallet.Account
+
 	current layout.Widget
 	dialog  layout.Widget
 	tabs    *decredmaterial.Tabs
 
+	addrs string
```
Comment (oluwandabira) [inline ui/window.go]: Remove. Window can track which account is selected instead

File: ui/window.go [hunk 11]
```diff
@@ -22,10 +22,18 @@ type Window struct {
 	wallet     *wallet.Wallet
 	walletInfo *wallet.MultiWalletInfo
 
+	selectedWallet  *wallet.InfoShort
+	selectedAccount *wallet.Account
```
Comment (oluwandabira) [inline ui/window.go]: Remove. Window already tracks the selected wallet

File: ui/recieve_page.go [hunk 12]
```diff
@@ -1 +1,301 @@
...
+				Top: unit.Dp(50),
+			}
+			inset.Layout(win.gtx, func() {
+				win.gtx.Constraints.Width.Min = 40
+				win.gtx.Constraints.Height.Min = 40
+				win.outputs.newAddress.Layout(win.gtx, &win.inputs.receiveIcons.newAddress)
+			})
+		}),
+	)
+}
```
Comment (oluwandabira) [inline ui/recieve_page.go]: see dialogs.go for how dialog's work with the window

File: ui/recieve_page.go [hunk 13]
```diff
@@ -1 +1,301 @@
...
+func (win *Window) setSelectedAccount(wallet wallet.InfoShort, account wallet.Account) {
+	win.selectedWallet = &wallet
+	win.selectedAccount = &account
+
+	win.outputs.selectedAccountNameLabel.Text = account.Name
+	win.outputs.selectedWalletNameLabel.Text = wallet.Name
+	win.outputs.selectedWalletBalLabel.Text = dcrutil.Amount(account.SpendableBalance).String()
+	win.outputs.selectedAccountBalanceLabel.Text = wallet.Balance
+	win.addrs = account.CurrentAddress
+}
```
Comment (oluwandabira) [inline ui/recieve_page.go]: Remove. The selected account just needs to be an int field of wallet indicating the selected account's index in `InfoShort.Accounts`

File: ui/recieve_page.go [hunk 14]
```diff
@@ -1 +1,301 @@
...
+				return
+			}
+			info.Accounts[i].CurrentAddress = addr
+			win.isNewAddrModal = false
+		}
+
+		win.setSelectedAccount(info, info.Accounts[i])
+		break
+	}
+}
```
Comment (oluwandabira) [inline ui/recieve_page.go]: Unnecessary

File: ui/recieve_page.go [hunk 15]
```diff
@@ -1 +1,301 @@
...
+				)
+			})
+		}),
+	)
+}
+
+func (win *Window) selectedAccountColumn() {
+	layout.Flex{}.Layout(win.gtx,
+		layout.Flexed(0.22, func() {
+		}),
```
Comment (oluwandabira) [inline ui/recieve_page.go]: Use the `win.combined.sel` widget. Should probably rename it to `accoutSel` first though

File: wallet/errors.go [hunk 16]
```diff
@@ -0,0 +1 @@
+package wallet
```
Comment (oluwandabira) [inline wallet/errors.go:1]: Is this file supposed to be here?

**Review by oluwandabira — CHANGES_REQUESTED** [review]

File: ui/widgets.go (hunk 8 above)
Comment (Sirmorrison) [reply ui/widgets.go]: So are you recommending i remove it?

File: ui/recieve_page.go (hunk 13 above)
Comment (Sirmorrison) [reply ui/recieve_page.go]: Are you referring to line 243 n 244?

File: ui/recieve_page.go (hunk 14 above)
Comment (Sirmorrison) [reply ui/recieve_page.go]: You may need to be more explicit.

File: ui/recieve_page.go (hunk 15 above)
Comment (Sirmorrison) [reply ui/recieve_page.go]: I dont follow.

File: wallet/errors.go (hunk 16 above)
Comment (Sirmorrison) [reply wallet/errors.go:1]: Think it got added after running go fmt ./...

File: ui/widgets.go (hunk 8 above)
Comment (oluwandabira) [reply ui/widgets.go]: Keep it if you want. It's really a design thing to have page titles but you'll probably have to have change the title for each page

File: ui/recieve_page.go (hunk 13 above)
Comment (oluwandabira) [reply ui/recieve_page.go]: I'm referring to the whole function. The setting of the selected account should be handled by the a change in the account Select widget

File: ui/recieve_page.go (hunk 14 above)
Comment (oluwandabira) [reply ui/recieve_page.go]: I'm saying the whole function is really unnecessary. What does default even mean in this context? why does it need to be set each time the page is laid out?. And clicks should be handled in `handle_input.go`

File: ui/recieve_page.go (hunk 15 above)
Comment (oluwandabira) [reply ui/recieve_page.go]: There is a `decredmaterial.Select` widget that in window (`win.combined.sel`). You can see how it looks and works on the send page. It allows showing and switching accounts based on the currently selected wallet. Also the toRecieve icon should be on the transaction page beside the toSend icon (for now).

File: wallet/errors.go (hunk 16 above)
Comment (oluwandabira) [reply wallet/errors.go:1]: A file go added by go fmt ?

File: ui/recieve_page.go (hunk 15 above)
Comment (Sirmorrison) [reply ui/recieve_page.go]: just to be sure, this is not the account selector function. this is just to display the selected account information. and is is shabbily implemented on the wallet page.

File: ui/decredmaterial/select.go [hunk 17]
```diff
@@ -10,7 +10,7 @@ type Select struct {
 	size     float32
 	changed  bool
 	open     bool
-	selected int
```
Comment (oluwandabira) [inline ui/decredmaterial/select.go:13]: Why is this changed?

File: ui/dialogs.go [hunk 18]
```diff
@@ -119,3 +121,107 @@ func (win *Window) AddAccountDiag() {
...
+			}
+			inset.Layout(win.gtx, func() {
+				win.gtx.Constraints.Width.Min = 40
+				win.gtx.Constraints.Height.Min = 40
+				win.outputs.newAddressDiag.Layout(win.gtx, &win.inputs.receiveIcons.newAddressDiag)
+			})
+		}),
+	)
+}
+
```
Comment (oluwandabira) [inline ui/dialogs.go]: Shouldn't be a dialog.

File: ui/dialogs.go [hunk 19]
```diff
@@ -119,3 +121,107 @@ func (win *Window) AddAccountDiag() {
...
+								)
+							})
+						}
+						decredmaterial.Card{}.Layout(win.gtx, selectedDetails)
+					}),
+				)
+			}),
+		)
+	})
+}
```
Comment (oluwandabira) [inline ui/dialogs.go]: I'm not sure where this gets displayed or why it would be necessary to display.

**Review by oluwandabira — CHANGES_REQUESTED** [review]

File: ui/decredmaterial/select.go (hunk 17 above)
Comment (Sirmorrison) [reply ui/decredmaterial/select.go:13]: Because i needed to write to it directly.

File: ui/dialogs.go (hunk 19 above)
Comment (Sirmorrison) [reply ui/dialogs.go]: should I remove it as it is part of the mockup in issue #7

Comment (oluwandabira) [reply ui/dialogs.go]: Does it get displayed?

File: ui/decredmaterial/select.go (hunk 17 above)
Comment (oluwandabira) [reply ui/decredmaterial/select.go:13]: I don't see why it would be necessary to change it in code when pretty much only the user changes it

File: ui/dialogs.go (hunk 19 above)
Comment (Sirmorrison) [reply ui/dialogs.go]: from the mobile app yes it is.

**Review by oluwandabira — APPROVED** [review]
